### PR TITLE
Chkfile initial guess problem in ROHF and ROKS (issue #1986)

### PR DESCRIPTION
**Pull request by sunqm**

### CODE REVIEW

**Review by fishjojo — APPROVED** [review]

File: pyscf/scf/hf.py [hunk 1]
```diff
@@ -656,12 +656,28 @@ def init_guess_by_chkfile(mol, chkfile_name, project=None):
...
+            the same molecule with different ordering of atoms.
+
     Returns:
         Density matrix, 2D ndarray
     '''
     from pyscf.scf import uhf
     dm = uhf.init_guess_by_chkfile(mol, chkfile_name, project)
-    return dm[0] + dm[1]
+    mo_coeff = dm.mo_coeff[0]
+    mo_occ = dm.mo_occ[0] + dm.mo_occ[1]
```
Comment (susilehtola) [inline pyscf/scf/hf.py:679]: Wouldn't e.g. UHF natural orbitals be a better way to initialize `mo_coeffs`?

File: pyscf/scf/rohf.py [hunk 2]
```diff
@@ -49,7 +49,28 @@ def init_guess_by_atom(mol):
...
+
+            If project is set to None, the projection is only applied when the
+            basis sets of the chkfile's molecule are different to the basis
+            sets of the given molecule (regardless whether the geometry of
+            the two molecules are different).  Note the basis sets are
+            considered to be different if the two molecules are derived from
+            the same molecule with different ordering of atoms.
+    '''
+    dm = uhf.init_guess_by_chkfile(mol, chkfile_name, project)
+    mo_coeff = dm.mo_coeff[0]
```
Comment (susilehtola) [inline pyscf/scf/rohf.py:71]: same here

File: pyscf/scf/hf.py (hunk 1 above)
Comment (sunqm) [reply pyscf/scf/hf.py]: Updated to natural orbitals here

File: pyscf/scf/rohf.py (hunk 2 above)
Comment (sunqm) [reply pyscf/scf/rohf.py:71]: Natural orbitals may be better. But I'm not sure how to properly assign the occupancy of spin-up and spin-down orbitals. I still use UHF spin-up orbitals here.

**Pull request merged**